libs/libc/ctype/lib_locale.c

```diff
@@ -0,0 +1,122 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <string.h>
+#include <ctype.h>
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+#if defined(__LIBCXXTOOLCHAIN__) && defined(__GNUC__)
+
+/* GNU libstdc++ is expecting an array _ctype_ to be defined. This array
+ * is usually defined in GNU libc or newlib.
+ * Here the system support only basic ASCII locale.
+ */
+
+const char _ctype_[] =
+{
+  0,
+  _C, _C, _C, _C, _C, _C, _C, _C, \
+  _C, _C | _S, _C | _S, _C | _S, _C | _S, _C | _S, _C, _C, \
+  _C, _C, _C, _C, _C, _C, _C, _C, \
+  _C, _C, _C, _C, _C, _C, _C, _C, \
+  _S | _B,  _P, _P, _P, _P, _P, _P, _P, \
+  _P, _P, _P, _P, _P, _P, _P, _P, \
+  _N, _N, _N, _N, _N, _N, _N, _N, \
+  _N, _N, _P, _P, _P, _P, _P, _P, \
+  _P, _U | _X, _U | _X, _U | _X, _U | _X, _U | _X, _U | _X, _U, \
+  _U, _U, _U, _U, _U, _U, _U, _U, \
+  _U, _U, _U, _U, _U, _U, _U, _U, \
+  _U, _U, _U, _P, _P, _P, _P, _P, \
+  _P, _L | _X, _L | _X, _L | _X, _L | _X, _L | _X, _L | _X, _L, \
+  _L, _L, _L, _L, _L, _L, _L, _L, \
+  _L, _L, _L, _L, _L, _L, _L, _L, \
+  _L, _L, _L, _P, _P, _P, _P, _C, \
+  0,  0,  0,  0,  0,  0,  0,  0, \
+  0,  0,  0,  0,  0,  0,  0,  0, \
+  0,  0,  0,  0,  0,  0,  0,  0, \
+  0,  0,  0,  0,  0,  0,  0,  0, \
+  0,  0,  0,  0,  0,  0,  0,  0, \
+  0,  0,  0,  0,  0,  0,  0,  0, \
+  0,  0,  0,  0,  0,  0,  0,  0, \
+  0,  0,  0,  0,  0,  0,  0,  0, \
+  0,  0,  0,  0,  0,  0,  0,  0, \
+  0,  0,  0,  0,  0,  0,  0,  0, \
+  0,  0,  0,  0,  0,  0,  0,  0, \
+  0,  0,  0,  0,  0,  0,  0,  0, \
+  0,  0,  0,  0,  0,  0,  0,  0, \
+  0,  0,  0,  0,  0,  0,  0,  0, \
+  0,  0,  0,  0,  0,  0,  0,  0, \
+  0,  0,  0,  0,  0,  0,  0,  0
+};
+
+#endif
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+#if defined(__LIBCXXTOOLCHAIN__) && defined(__GNUC__)
+
+/* GNU libstdc++ is expecting some functions to be defined.
+ */
+
+/* Shall return the max number of bytes of a multibyte character */
+
+int __locale_mb_cur_max(void)
+{
+  return 1;
+}
+
+#endif
+
+char *setlocale(int category, const char *locale)
+{
+  if (strcmp (locale, "POSIX") &&
+      strcmp (locale, "C")
+      && strcmp (locale, ""))
+    {
+      return NULL;
+    }
+
+  return "C";
+}
+
+int strcoll(const char *first, const char *second)
```

Comment:
already implement in lib_strcoll.c

Reply:
OK but that's strange: theses symbols are missing if I don't define them... Is there any option that I am missing?

Reply:
OK I missed a config on my side: I'll enabled the minimum locale support of NuttX if C++ toolchain is enabled.